File: prisma/scraper.js
const puppeteer = require("puppeteer");
const fs = require("fs");

(async () => {
    const url = "https://jacklich10.com/bigboard/nfl/";
    const browser = await puppeteer.launch({ headless: true });
    const page = await browser.newPage();
    
    await page.goto(url, { waitUntil: "networkidle2" });

    const players = await page.evaluate(() => {
        const rows = document.querySelectorAll(".rt-tr-group"); // Select each row in the table
        let data = [];

        rows.forEach(row => {
            try {
                const ranking = row.querySelector(".rt-td:nth-child(1) .rt-td-inner")?.innerText.trim();
                const schoolLogo = row.querySelector(".rt-td:nth-child(2) img")?.src;
                const playerImage = row.querySelector(".rt-td:nth-child(3) img")?.src;
                const position = row.querySelector(".rt-td:nth-child(4) span:nth-child(1)")?.innerText.trim();
                const year = row.querySelector(".rt-td:nth-child(4) span:nth-child(2)")?.innerText.trim();
                const firstname = row.querySelector(".rt-td:nth-child(5) div div span:nth-child(1)")?.innerText.trim();
                const lastname = row.querySelector(".rt-td:nth-child(5) div div span:nth-child(2)")?.innerText.trim();

                // Extract school name from logo URL (custom logic, update if needed)
                let school = "Unknown";
                if (schoolLogo) {
                    const parts = schoolLogo.split("/");
                    school = parts[parts.length - 1].replace(".png", "");
                }

                if (firstname && lastname) {
                    data.push({
                        firstname,
                        lastname,
                        ranking: parseInt(ranking, 10),
                        school,
                        position,
                        schoolLogo,
                        playerImage,
                        year,
                    });
                }
            } catch (error) {
                console.log("Error parsing row:", error);
            }
        });

        return data;
    });

    // Save data to JSON file
    fs.writeFileSync("players.json", JSON.stringify(players, null, 2));
    console.log("Scraping completed. Data saved in players.json");

    await browser.close();
})();

File: src/scraper.py
import requests
from bs4 import BeautifulSoup
import json

# URL of the page to scrape
url = 'https://jacklich10.com/bigboard/nfl/'

# Send a GET request to the URL
response = requests.get(url)
# Parse the HTML content using BeautifulSoup
soup = BeautifulSoup(response.content, 'html.parser')

# Find all player rows
player_rows = soup.find_all('div', class_='rt-tr-group')
print(player_rows)

# List to store player information
players = []

# Iterate over each player row and extract the required information
for row in player_rows:
    player_info = row.find('div', class_='rt-tr')
    if player_info:
        ranking = player_info.find('div', class_='rt-td-inner').text.strip()
        school_logo = player_info.find_all('img')[0]['src']
        player_image = player_info.find_all('img')[1]['src']
        position = player_info.find_all('div', class_='rt-td-inner')[3].find('span').text.strip()
        year = player_info.find_all('div', class_='rt-td-inner')[3].find_all('span')[1].text.strip()
        firstname = player_info.find_all('div', class_='rt-td-inner')[4].find('span').text.strip()
        lastname = player_info.find_all('div', class_='rt-td-inner')[4].find_all('span')[1].text.strip()
        school = player_image.split('/')[-1].split('-')[0]

        # Append the extracted information to the players list
        players.append({
            'firstname': firstname,
            'lastname': lastname,
            'ranking': int(ranking),
            'school': school,
            'position': position,
            'schoolLogo': school_logo,
            'playerImage': player_image,
            'year': year,
        })

# Convert the players list to JSON format
players_json = json.dumps(players, indent=4)

# Print the JSON output
print(players_json)
